driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -393,8 +393,13 @@ private MongoIncompatibleDriverException createIncompatibleException(final Clust
                     MIN_DRIVER_WIRE_VERSION, MIN_DRIVER_SERVER_VERSION);
         } else {
             incompatibleServer = curDescription.findServerIncompatiblyNewerThanDriver();
-            message = format("Server at %s requires wire version %d, but this version of the driver only supports up to %d.",
-                    incompatibleServer.getAddress(), incompatibleServer.getMinWireVersion(), MAX_DRIVER_WIRE_VERSION);
+            if (incompatibleServer != null) {
+                message = format("Server at %s requires wire version %d, but this version of the driver only supports up to %d.",
+                        incompatibleServer.getAddress(), incompatibleServer.getMinWireVersion(), MAX_DRIVER_WIRE_VERSION);
+            } else {
+                message = format("Incompatible server. This version of the driver requires at least wire version %d (MongoDB %s) up to %d.",
```

Comment:
We shouldn't hit this clause but as `findServerIncompatiblyNewerThanDriver` can return null this seemed the cleanest way to handle that possibility.